modules/store/testing/src/testing.ts

```diff
@@ -18,22 +22,115 @@ export interface MockStoreConfig<T> {
   selectors?: MockSelector[];
 }
 
+/**
+ * @description
+ * Creates mock store providers.
+ *
+ * @param config `MockStoreConfig<T>` to provide the values for `INITIAL_STATE` and `MOCK_SELECTORS` tokens.
+ * By default, `initialState` and `selectors` are not defined.
+ *
+ * @usageNotes
+ *
+ * **With `TestBed.configureTestingModule`**
+ *
+ * ```ts
+ * describe('Books Component', () => {
+ *   let store: MockStore;
+ *
+ *   beforeEach(() => {
+ *     TestBed.configureTestingModule({
+ *       providers: [
+ *         provideMockStore({
+ *           initialState: { books: { entities: [] } },
+ *           selectors: [
+ *             { selector: selectAllBooks, value: ['Book 1', 'Book 2'] },
+ *             { selector: selectVisibleBooks, value: ['Book 1'] },
+ *           ],
+ *         }),
+ *       ],
+ *     });
+ *
+ *     store = TestBed.inject(MockStore);
+ *   });
+ * });
+ * ```
+ *
+ * **With `Injector.create`**
+ *
+ * ```ts
+ * describe('Counter Component', () => {
+ *   let injector: Injector;
+ *   let store: MockStore;
+ *
+ *   beforeEach(() => {
+ *     injector = Injector.create({
+ *       providers: [
+ *         provideMockStore({ initialState: { counter: 0 } }),
+ *       ],
+ *     });
+ *     store = injector.get(MockStore);
+ *   });
+ * });
+ * ```
+ */
 export function provideMockStore<T = any>(
   config: MockStoreConfig<T> = {}
-): Provider[] {
+): (ValueProvider | ExistingProvider | FactoryProvider)[] {
   setNgrxMockEnvironment(true);
   return [
-    ActionsSubject,
-    MockState,
-    MockStore,
     { provide: INITIAL_STATE, useValue: config.initialState || {} },
     { provide: MOCK_SELECTORS, useValue: config.selectors },
-    { provide: StateObservable, useClass: MockState },
-    { provide: ReducerManager, useClass: MockReducerManager },
+    {
+      provide: ActionsSubject,
+      useFactory: () => new ActionsSubject(),
+      deps: [],
+    },
+    {
+      provide: MockState,
+      useFactory: () => new MockState<T>(),
+      deps: [],
+    },
+    {
+      provide: StateObservable,
+      useFactory: () => new MockState<T>(),
+      deps: [],
+    },
+    {
+      provide: ReducerManager,
+      useFactory: () => new MockReducerManager(),
+      deps: [],
+    },
+    {
+      provide: MockStore,
+      useFactory: mockStoreFactory,
+      deps: [
+        MockState,
+        ActionsSubject,
+        ReducerManager,
+        INITIAL_STATE,
+        MOCK_SELECTORS,
+      ],
+    },
     { provide: Store, useExisting: MockStore },
   ];
 }
 
+function mockStoreFactory<T>(
+  mockState: MockState<T>,
+  actionsSubject: ActionsSubject,
+  reducerManager: ReducerManager,
+  initialState: T,
+  mockSelectors: MockSelector[]
+): MockStore<T> {
+  return new MockStore(
+    mockState,
+    actionsSubject,
+    reducerManager,
+    initialState,
+    mockSelectors
+  );
+}
+
```

Comment:
Instead of going through the `Injector` every time, maybe let's create the `getMockStore` or `createMockStore` function:

```suggestion
export function getMockStore<T>(
  config: MockStoreConfig<T> = {}
): MockStore<T> {
  return Injector.create({ providers: [ provideMockStore(config) ]}).get(MockStore);
}
```

Reply:
Done